etc/scripts/github_release.py

```diff
@@ -0,0 +1,158 @@
+
+import argparse
+from commoncode.fileutils import resource_iter
+import fnmatch
+import os
+import subprocess
+import sys
+
+python_version = str(sys.version_info[0]) + str(sys.version_info[1])
+py_abi = "{0}cp{1}{0}".format("*", python_version)
+
+
+def release_asset(token, tag, repo, body, user, limit, file):
+    """
+    Release .whl,.ABOUT,.NOTICE,.LICENSE to github repository from 
+    given directory.
+    """
+
+    os.environ["GITHUB_TOKEN"] = token
+    thirdparty = list(resource_iter(file, with_dirs=False))
+    dependencies = [
+        files
+        for files in thirdparty
+        if fnmatch.fnmatchcase(files, "*py3*")
+        or fnmatch.fnmatchcase(files, py_abi)
+        or (
+            fnmatch.fnmatchcase(files, "*tar.gz*")
+            and not fnmatch.fnmatchcase(files, "*py2-ipaddress-3.4.1.tar.gz*")
+        )
+    ]
+    for deps in dependencies:
+        print(deps)
+        subprocess.run(
+            [
+                "python3",
+                "-m",
+                "github_release_retry.github_release_retry",
+                "--user",
+                user,
+                "--repo",
+                repo,
+                "--tag_name",
+                tag,
+                "--body_string",
+                body,
+                "--retry_limit",
+                limit,
+                deps,
+            ]
+        )
+
+
+def main_with_args(args: str) -> None:
+    parser = argparse.ArgumentParser(
+        description="""Creates a GitHub release (if it does not already exist) and uploads files to the release.
+Please pass the GITHUB_TOKEN as an argument.
+EXAMPLE:
+github-release-asset \\
+  --user Abhishek-Dev09 \\
+  --repo thirdparty \\
+  --tag_name v1.0 \\
+  --body_string "Python 3.6 wheels" \\
+  hello-world.zip RELEASE_NOTES.txt
+""",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--user",
+        help="Required: The GitHub username or organization name in which the repo resides. ",
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        "--token",
+        help="Required: The Github token is required to acess the repository where you want to upload . ",
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        "--repo",
+        help="Required: The GitHub repo name in which to make the release. ",
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        "--tag_name",
+        help="Required: The name of the tag to create or use. ",
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        "--body_string",
+        help="Required (or use --body_file): Text describing the release. Ignored if the release already exists.",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--retry_limit",
+        help="The number of times to retry creating/getting the release and/or uploading each file. ",
```

Comment:
You have a trailing space in the help string here and most everywhere above.

Reply:
removing now